File: docs/guides/go/src/abTestImplementationChecklist.go
package main

import (
	"github.com/algolia/algoliasearch-client-go/v4/algolia/search"
)

func getUserToken() (string, error) {
	// Implement your logic here
	return "", nil
}

func abTestImplementationChecklist() {
	client, err := search.NewClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")
	if err != nil {
		// The client can fail to initialize if you pass an invalid parameter.
		panic(err)
	}

	// Set the searchParams and get the current user token
	userToken, err := getUserToken()
	if err != nil {
		panic(err)
	}

	var searchParams *search.SearchParams

	// Is the user token anonymous?
	if userToken == "" || userToken == "YOUR_ANONYMOUS_USER_TOKEN" {
		// Disable A/B testing for this request
		searchParams = search.SearchParamsObjectAsSearchParams(
			search.NewSearchParamsObject().
				SetQuery("User search query").
				SetEnableABTest(false),
		)
	} else {
		// Set the user token to the current user token
		searchParams = search.SearchParamsObjectAsSearchParams(
			search.NewSearchParamsObject().
				SetQuery("User search query").
				SetEnableABTest(true).
				SetUserToken(userToken),
		)
	}

	// Perform the searchSingleIndex
	_, err = client.SearchSingleIndex(client.NewApiSearchSingleIndexRequest(
		"<YOUR_INDEX_NAME>").WithSearchParams(searchParams))
	if err != nil {
		// SearchSingleIndex errors
		panic(err)
	}
}

File: docs/guides/dart/lib/abTestImplementationChecklist.dart
import 'package:algolia_client_search/algolia_client_search.dart';

String getUserToken() {
  // Implement your logic here
  return "";
}

void abTestImplementationChecklist() async {
  final client =
      SearchClient(appId: 'ALGOLIA_APPLICATION_ID', apiKey: 'ALGOLIA_API_KEY');

  // Set the query and get the current user token
  final query = "User search query";
  final userToken = getUserToken();

  // Set the searchParams
  final SearchParamsObject searchParams;

  // Is the user token anonymous?
  if (userToken.isEmpty || userToken == "YOUR_ANONYMOUS_USER_TOKEN") {
    // Disable A/B testing for this request
    searchParams = SearchParamsObject(
      query: query,
      enableABTest: false,
    );
  } else {
    // Set the user token to the current user token
    searchParams = SearchParamsObject(
      query: query,
      userToken: userToken,
    );
  }

  try {
    // Perform the searchSingleIndex
    final result = await client.searchSingleIndex(
      indexName: "<YOUR_INDEX_NAME>",
      searchParams: searchParams,
    );
    // SearchSingleIndex results
    print(result);
  } catch (err) {
    // SearchSingleIndex errors
    print(err);
  }
}
